### ISSUE TITLE
`dotnet tool install -a arm64` fails

### ISSUE THREAD

**Issue by richlander** [dotnet/sdk#21395]
Same thing as #21393 (but different command).

```bash
rich@MacBook-Air-M1-2020 app4321 % dotnet --info | grep RID
 RID:         osx-x64
rich@MacBook-Air-M1-2020 app4321 % dotnet tool install -g app4321 --add-source bin/Debug -a arm64
System.AggregateException: One or more errors occurred. (Resolving the current runtime identifier failed.)
 ---> Microsoft.DotNet.Cli.Utils.GracefulException: Resolving the current runtime identifier failed.
   at Microsoft.DotNet.Cli.CommonOptions.GetCurrentRuntimeId()
   at Microsoft.DotNet.Cli.CommonOptions.ResolveRidShorthandOptionsToRuntimeIdentifier(String os, String arch)
   at Microsoft.DotNet.ShellShim.ShellShimTemplateFinder.ResolveAppHostSourceDirectoryAsync(String archOption, String targetFramework)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result()
   at Microsoft.DotNet.Tools.Tool.Install.ToolInstallGlobalOrToolPathCommand.Execute()
   at Microsoft.DotNet.Tools.Tool.Install.ToolInstallCommand.Execute()
   at Microsoft.DotNet.Cli.DotNetTopLevelCommandBase.RunCommand(String[] args)
   at Microsoft.DotNet.Tools.Tool.ToolCommand.Run(String[] args)
   at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, TimeSpan startupTime, ITelemetry telemetryClient)
   at Microsoft.DotNet.Cli.Program.Main(String[] args)
```

**Comment by sfoslund** [comment]
@richlander Does the file `dotnet/sdk/{version}/NETCoreSdkRuntimeIdentifierChain.txt` exist in your installation? If so, what's the content?

**Comment by richlander** [comment]
```bash
rich@MacBook-Air-M1-2020 app4321 % dotnet --info | grep RID
 RID:         osx-x64
rich@MacBook-Air-M1-2020 app4321 % cat /usr/local/share/dotnet/x64/sdk/6.0.100-rc.2.21473.35/NETCoreSdkRuntimeIdentifierChain.txt 
osx-x64
osx
unix-x64
unix
any
base%                                                                           
rich@MacBook-Air-M1-2020 app4321 % cat /usr/local/share/dotnet/sdk/6.0.100-rc.2.21473.35/NETCoreSdkRuntimeIdentifierChain.txt 
osx-arm64
osx
unix-arm64
unix
any
base%                                                   
```

**Comment by sfoslund** [comment]
Hmm it seems to be failing to find that file: https://github.com/dotnet/sdk/blob/main/src/Cli/dotnet/CommonOptions.cs#L205, I'll try to repro

**Comment by sfoslund** [comment]
Fixed by https://github.com/dotnet/sdk/pull/21435

**Comment by richlander** [comment]
Just validated this fix in `6.0.100-rc.2.21478.25`. Thanks!
